c7n/resources/account.py

```diff
@@ -424,6 +425,86 @@ def process(self, resources, event=None):
         return []
 
 
+@actions.register('set-password-policy')
+class SetAccountPasswordPolicy(BaseAction):
+    """Set an account's password policy.
+
+    This only changes the policy for the items provided.
+    If this is the first time setting a password policy and an item is not provided it will be
+    set to the defaults defined in the boto docs for IAM.Client.update_account_password_policy
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-account-password-policy
+                resource: account
+                filters:
+                  - not:
+                    - type: password-policy
+                      key: MinimumPasswordLength
+                      value: 10
+                      op: ge
+                actions:
+                    - type: set-password-policy
+                      policy:
+                        - key: MinimumPasswordLength
+                          value: 20
+    """
+    schema = type_schema(
+        'set-password-policy',
+        policy={
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'key': {'type': 'string'},
+                    'value': {'oneOf': [{'type': 'integer'}, {'type': 'boolean'}]}
+                }
+            }
+        })
+    shape = 'PasswordPolicy'
+    permissions = ('iam:GetAccountPasswordPolicy', 'iam:UpdateAccountPasswordPolicy')
+    # We need to whitelist the values accepted by the update call as it is not available via shape
+    policy_whitelist = (
+        "MinimumPasswordLength", "RequireSymbols", "RequireNumbers", "RequireUppercaseCharacters",
+        "RequireLowercaseCharacters", "AllowUsersToChangePassword", "MaxPasswordAge",
+        "PasswordReusePrevention", "HardExpiry"
+    )
+
+    def validate(self):
+        policy = dict({item['key']: item['value'] for item in self.data['policy']})
+        attrs = policy
```

Comment:
why the redefinition, seems superflous

Reply:
Ah, great question. This is superflous. Will update in a few.

Reply:
Updated in b382acd.